lapis2-docs/src/components/ConfigGenerator/configContext.tsx

```diff
@@ -24,49 +24,37 @@ export type MetadataType = z.infer<typeof metadataTypeSchema>;
 export const metadataSchema = z.object({
     name: z.string(),
     type: metadataTypeSchema,
-    generateIndex: z.boolean(),
+    generateIndex: z.boolean().optional(),
     autocomplete: z.boolean().optional(),
     required: z.boolean().optional(),
     notSearchable: z.boolean().optional(),
 });
 
-export type Metadata = {
-    name: string;
-    type: MetadataType;
-    generateIndex: boolean;
-    autocomplete?: boolean;
-    required?: boolean;
-    notSearchable?: boolean;
-};
+export type Metadata = z.infer<typeof metadataSchema>;
 
 export const featureSchema = z.object({
     name: z.string(),
 });
 export type Feature = z.infer<typeof featureSchema>;
 
-export const ConfigSchema = z.object({
+export const schemaSchema = z.object({
     instanceName: z.string(),
     opennessLevel: opennessLevelSchema,
     metadata: z.array(metadataSchema),
     primaryKey: z.string(),
     dateToSortBy: z.string(),
     partitionBy: z.string(),
-    tableColumns: z.array(z.string()),
+    tableColumns: z.array(z.string()).optional(),
     features: z.array(featureSchema),
 });
 
-export type Config = {
-    instanceName: string;
-    opennessLevel: OpennessLevel;
-    metadata: Metadata[];
-    primaryKey: string;
-    dateToSortBy: string;
-    partitionBy: string;
-    tableColumns: string[];
-    features: Feature[];
-};
+export type Schema = z.infer<typeof schemaSchema>;
```

Comment:
;) schemaSchema

Reply:
Well, the schema in this file is: call the zod schema `somethingSchema` and the type `Something`. If you have a better suggestion than `schemaSchema`, I'm happy to hear it :D

Reply:
I have no better idea, but funny nonetheless. ;)